lmdeploy/serve/async_engine.py

```diff
@@ -476,8 +475,8 @@ async def gather():
                 ])
                 outputs.put(None)
 
-            proc = Thread(
-                target=lambda: self.loop.run_until_complete(gather()))
+            proc = Thread(target=lambda: asyncio.new_event_loop().
```

Comment:
Why do you use thread instead of coroutine here?
`asyncio.get_event_loop().create_task` would give create a new coroutine task parallelized with the current coroutine. and `asyncio.Queue` is awaitable.

Reply:
> Why do you use thread instead of coroutine here? `asyncio.get_event_loop().create_task` would give create a new coroutine task parallelized with the current coroutine. and `asyncio.Queue` is awaitable.

Just want to yield items without async. Do you have demo codes?

Reply:
`Thread` -> `asyncio.get_event_loop().create_task`
`Queue` -> `asyncio.queue`

Reply:
https://github.com/grimoire/lmdeploy/blob/d75c757f1712ffb1a0958758e1e0d4b18743ca8d/lmdeploy/pytorch/engine/engine.py#L718

Reply:
How can I return a generator for stream infer?
https://github.com/InternLM/lmdeploy/blob/3d355b5430acc37c284c62c1cd174e84780ea2be/lmdeploy/serve/async_engine.py#L484-L488

Reply:
```
 try: 
	     out = await outputs.get() 
	     if out is None: 
	         break 
	     yield out 
```

Reply:
> ```
>  try: 
> 	     out = await outputs.get() 
> 	     if out is None: 
> 	         break 
> 	     yield out 
> ```

but this is not a async function.

Reply:
https://github.com/InternLM/lmdeploy/blob/3d355b5430acc37c284c62c1cd174e84780ea2be/lmdeploy/pytorch/engine/engine.py#L1103

Reply:
To avoid the performance drop, I used the original `Thread` and `Queue`.